test/infrastructure/docker/docker/loadbalancer.go

```diff
@@ -65,14 +67,41 @@ func NewLoadBalancer(cluster *clusterv1.Cluster) (*LoadBalancer, error) {
 		return nil, fmt.Errorf("create load balancer: %s", err)
 	}
 
+	image := getLoadBalancerImage(dockerCluster)
+
 	return &LoadBalancer{
 		name:      cluster.Name,
+		image:     image,
 		container: container,
 		ipFamily:  ipFamily,
 		lbCreator: &Manager{},
 	}, nil
 }
 
+// getLoadBalancerImage will return the image (e.g. "kindest/haproxy:2.1.1-alpine") to use for
+// the load balancer.
+func getLoadBalancerImage(dockerCluster *v1alpha4.DockerCluster) string {
+	// Check if a non-default image was provided
+	image := loadbalancer.Image
+	imageRepo := ""
+	imageTag := ""
+
+	if dockerCluster != nil {
+		imageRepo = dockerCluster.Spec.LoadBalancer.ImageRepository
+		imageTag = dockerCluster.Spec.LoadBalancer.ImageTag
+	}
+
+	if imageRepo == "" {
+		imageRepo = loadbalancer.ImageRepository
```

Comment:
You can also already set the defaults in l.86 (+ l.87)

Reply:
Done